internal/hud/renderer.go

```diff
@@ -319,6 +320,13 @@ func (r *Renderer) SetUp() (chan tcell.Event, error) {
 	return screenEvents, nil
 }
 
+func (r Renderer) RTY() rty.RTY {
```

Comment:
ok, I see the problem. 

There's a bug in your RTY() function. This needs to be `(r *Renderer)` instead of `(r Renderer)`. Right now it's making a copy of Renderer when you call the function, which is not what you want.

once you fix that, you can remove the rendundant RLock() that we talked about above

Reply:
@nicks Oh yes sorry for that ! Thank you.